src/Curry.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
```

Comment:
Ah ignore me, that was a terrible test that missed the whole point because `'1'` was being type juggled on the way in. :)